platforms/documentation/docs/src/docs/userguide/dep-man/basics/dependency_resolution.adoc

```diff
@@ -0,0 +1,204 @@
+Each *variant* is tied to a specific *component* version.
+
+For a more detailed view of which *variant* Gradle resolved for the `runtimeClasspath` configuration, you can run:
+
+[source,text]
+----
+$ ./gradlew :app:dependencyInsight --configuration runtimeClasspath --dependency com.google.guava:guava:33.0.0-jre
+
+> Task :app:dependencyInsight
+
+com.google.guava:guava:33.0.0-jre
+  Variant jreRuntimeElements:
+    | Attribute Name                 | Provided     | Requested    |
+    |--------------------------------|--------------|--------------|
+    | org.gradle.status              | release      |              |
+    | org.gradle.category            | library      | library      |
+    | org.gradle.dependency.bundling | external     | external     |
+    | org.gradle.jvm.environment     | standard-jvm | standard-jvm |
+    | org.gradle.jvm.version         | 8            | 11           |
+    | org.gradle.libraryelements     | jar          | jar          |
+    | org.gradle.usage               | java-runtime | java-runtime |
+
+com.google.guava:guava:33.0.0-jre
+\--- runtimeClasspath
+----
+
+In this example, Gradle uses the `jreRuntimeElements` variant* of `guava` for the `runtimeClasspath` configuration.
+
+== 2. Artifact Resolution
+
+**Artifact resolution** occurs after the dependency graph is constructed.
+It focuses on fetching the necessary physical files (*artifacts*) for the build and ensuring they are available, valid, and correctly versioned.
+
+This process uses the resolved graph and repository definitions to produce the required files as output.
+
+=== 1. Fetching artifacts
+
+Gradle locates and downloads the actual artifacts (such as JAR files, ZIP files, etc.) required by the project.
+These artifacts correspond to the dependencies identified during graph resolution.
+
+In our example, Gradle resolved the `jreRuntimeElements` variant of `com.google.guava:guava:33.0.0-jre` during the dependency graph resolution.
+The artifact resolution phase retrieves the corresponding JAR file (`guava-33.0.0-jre.jar`) and its transitive dependencies, like `jsr305` or `failureaccess`, from the specified repositories.
+
+=== 2. Artifact resolution rules
+
+If any artifact resolution rules are defined in the build script, they are applied at this stage.
+These rules might involve substituting one artifact for another or enforcing specific artifact versions.
+
+In our example, if the build script included a rule to substitute `com.google.guava:guava:33.0.0-jre` with a custom-built version, Gradle would apply this rule during artifact resolution, ensuring the correct version is used according to the project's specific needs.
+
+== Summary
+
+The dependency resolution engine is variant-aware and selects one or more *variants* of each *component* based on your build requirements.
+It may fail if the *variant* selection result is ambiguous, meaning that Gradle does not have enough information to select one of multiple mutual exclusive variants.
```

Comment:
```suggestion
It may fail if the *variant* selection result is ambiguous, meaning that Gradle does not have enough information to select one of multiple variants.
```

Not all variants of a component are mutually exclusive. Depending on the capabilities of the variants in a component, it is possible to select two variants of a single component

Reply:
Removed. This was a leftover note.